sync_engagement_db_to_coda.py

```diff
@@ -43,4 +44,5 @@
     engagement_db = pipeline_config.engagement_database.init_engagement_db_client(google_cloud_credentials_file_path)
     coda = pipeline_config.coda_sync.coda.init_coda_client(google_cloud_credentials_file_path)
 
+    ensure_coda_datasets_up_to_date(coda, pipeline_config.coda_sync.sync_config, google_cloud_credentials_file_path)
```

Comment:
Should we sync twice in a pipeline run or just once e.g only during engagement_db_to_coda sync?

Reply:
In a pipeline run it should be once but that's not the convenient way to implement because each script can be run independently and they should ensure coda datasets are up to date before sync. As discussed earlier I'm going to experiment this before I put in a separate manually triggered tool. https://github.com/AfricasVoices/Test-Pipeline-Engagement-DB/pull/125#discussion_r742045024